sympy/solvers/tests/test_solveset.py

```diff
@@ -752,12 +752,13 @@ def test_solve_abs():
     eqab = eq.subs(reps)
     for si in sol.subs(reps):
         assert not eqab.subs(x, si)
-    assert dumeq(solveset(Eq(sin(Abs(x)), 1), x, domain=S.Reals), Union(
-        Intersection(Interval(0, oo), Union(
-        Intersection(ImageSet(Lambda(n, 2*n*pi + 3*pi/2), S.Integers),
-            Interval(-oo, 0)),
-        Intersection(ImageSet(Lambda(n, 2*n*pi + pi/2), S.Integers),
-            Interval(0, oo))))))
+
+    eq = Eq(sin(Abs(x)), 1)
+    sol = Union(
+        ImageSet(Lambda(n, -2*n*pi - pi/2), Range(0, oo, 1)),
+        ImageSet(Lambda(n, 2*n*pi + pi/2), Range(0, oo, 1))
+    )
+    assert dumeq(solveset(eq, x, domain=S.Reals), sol)
```

Comment:
The previous answer here evaluates to:
```python
In [1]: Union(
   ...:         Intersection(Interval(0, oo), Union(
   ...:         Intersection(ImageSet(Lambda(n, 2*n*pi + 3*pi/2), S.Integers),
   ...:             Interval(-oo, 0)),
   ...:         Intersection(ImageSet(Lambda(n, 2*n*pi + pi/2), S.Integers),
   ...:             Interval(0, oo)))))
Out[1]: 
         ⎧        π │      ⎫
[0, ∞) ∩ ⎨2⋅π⋅n + ─ │ n ∊ ℤ⎬
         ⎩        2 │      ⎭
```
This is incorrect because there also negative solutions for `x`. This is due to a bug in the intersection handler for imageset and interval that is fixed here so that the answer is:
```python
In [2]: solveset(Eq(sin(Abs(x)), 1), x, domain=S.Reals)
Out[2]: 
⎧         π │              ⎫   ⎧        π │              ⎫
⎨-2⋅n⋅π - ─ │ n ∊ {0, 1, …}⎬ ∪ ⎨2⋅n⋅π + ─ │ n ∊ {0, 1, …}⎬
⎩         2 │              ⎭   ⎩        2 │              ⎭
```